app/(candidate)/dashboard/website/editor/components/WebsiteCreateFlow.js

```diff
@@ -0,0 +1,308 @@
+    }
+
+    return resp.ok
+  }
+
+  function handleStepChange(newStep) {
+    setStep(newStep)
+  }
+
+  function handleVanityPathChange(value) {
+    setWebsite((current) => ({
+      ...current,
+      vanityPath: value,
+    }))
+  }
+
+  function handleLogoChange(file) {
+    if (file) {
+      const reader = new FileReader()
+      reader.onloadend = () => setLogo(reader.result, file)
+      reader.readAsDataURL(file)
+    } else {
+      setLogo(undefined, undefined)
+    }
+
+    function setLogo(url, file) {
+      setWebsite((current) => ({
+        ...current,
+        content: {
+          ...current.content,
+          logo: url,
+          logoFile: file,
+        },
+      }))
+    }
+  }
+
+  function handleThemeChange(value) {
+    setWebsite((current) => ({
+      ...current,
+      content: {
+        ...current.content,
+        theme: value,
+      },
+    }))
+  }
+
+  function handleTitleChange(value) {
+    setWebsite((current) => ({
+      ...current,
+      content: {
+        ...current.content,
+        main: { ...current.content.main, title: value },
+      },
+    }))
+  }
+
+  function handleTaglineChange(value) {
+    setWebsite((current) => ({
+      ...current,
+      content: {
+        ...current.content,
+        main: { ...current.content.main, tagline: value },
+      },
+    }))
+  }
+
+  function handleHeroChange(file) {
+    if (file) {
+      const reader = new FileReader()
+      reader.onloadend = () => setHero(reader.result, file)
+      reader.readAsDataURL(file)
+    } else {
+      setHero(undefined, undefined)
+    }
+
+    function setHero(url, file) {
+      setWebsite((current) => ({
+        ...current,
+        content: {
+          ...current.content,
+          main: { ...current.content.main, image: url },
+          heroFile: file,
+        },
+      }))
+    }
+  }
+
+  function handleBioChange(value) {
+    setWebsite((current) => ({
+      ...current,
+      content: {
+        ...current.content,
+        about: { ...current.content.about, bio: value },
+      },
+    }))
+  }
+
+  function handleIssuesChange(issues) {
+    setWebsite((current) => ({
+      ...current,
+      content: {
+        ...current.content,
+        about: { ...current.content.about, issues },
+      },
+    }))
+  }
+
+  function handleAddressChange(value) {
+    setWebsite((current) => ({
+      ...current,
+      content: {
+        ...current.content,
+        contact: { ...current.content.contact, address: value },
+      },
+    }))
+  }
+
+  function handleEmailChange(value) {
+    setWebsite((current) => ({
+      ...current,
+      content: {
+        ...current.content,
+        contact: { ...current.content.contact, email: value },
+      },
+    }))
+  }
+
+  function handlePhoneChange(value) {
+    setWebsite((current) => ({
+      ...current,
+      content: {
+        ...current.content,
+        contact: { ...current.content.contact, phone: value },
+      },
+    }))
```

Comment:
You can probably keep this more DRY by passing two more parameters - one for the content key and one for the sub key.